iocore/net/quic/QUICFrameReactor.h

```diff
@@ -0,0 +1,63 @@
+
+#pragma once
+
+#include "QUICFrame.h"
+
+class QUICFrameReactorEvent;
+class QUICAckReceivedReactor;
+using QUICFrameReactorEventUPtr = std::unique_ptr<QUICFrameReactorEvent>;
+
+class QUICAckReceivedReactor
+{
+public:
+  virtual void
+  on_frame_acked(QUICFrameUPtr &frame, QUICEncryptionLevel level)
+  {
+  }
+  virtual void
+  on_frame_lost(QUICFrameUPtr &frame, QUICEncryptionLevel level)
+  {
+  }
+};
+
+class QUICFrameReactorEvent
+{
+public:
+  QUICFrameReactorEvent(QUICAckReceivedReactor *reactor, QUICFrameUPtr &frame, QUICEncryptionLevel level)
+    : _frame(std::move(frame)), _reactor(reactor), _level(level)
```

Comment:
It's just an idea but adding `generated_by()` or `owner()` that returns an object responsible for a frame to QUICFrame might be another option, I'm not sure which design is better yet though.